distributed/protocol/utils.py

```diff
@@ -55,6 +55,10 @@ def merge_frames(header, frames):
     if not frames:
         return frames
 
+    if any(x.__class__.__name__ == 'MemoryPointer' for x in frames):
+        # XXX
+        return frames
```

Comment:
@TomAugspurger can you expand on the issue here?  I'm just curious if there is something we can start up now in parallel so that this can go away over time.

Reply:
I was in the midst of fixing this when baby came. https://github.com/TomAugspurger/distributed/blob/1307616d513cec751eca6bc3a9a267b5871ee127/distributed/protocol/utils.py#L67-L85 has the "fixed" version.

That fix currently relies Numba DeviceNDArray as a stand-in for a gpu-memoryview. I would recommend that we (dask / rapids / pydata community members) define a lightweight object that implements `__cuda_array_interface__`, which both distributed and ucx-py can rely on.

Reply:
This is fixing https://github.com/dask/distributed/issues/2565

Reply:
And looking at my local directory, I have an uncommitted `import pdb; pdb.set_trace()`, so this likely isn't complete done :)

@quasiben this may be worth looking into before that segfault we spoke about earlier. It's possible (but not certain) that this fixes it.

Reply:
Thanks for the tip!

Reply:
Is there a test that triggers this?  Or, more directly, when does this come up?  cc @TomAugspurger